src/expr/node.h

```diff
@@ -923,24 +923,100 @@ inline std::ostream& operator<<(std::ostream& out, TNode n) {
   return out;
 }
 
+namespace {
+
+template <typename T>
+void nodeContainerToOut(std::ostream& out, const T& container)
+{
+  out << "[";
+  bool is_first = true;
+  for (auto item : container)
```

Comment:
Any reason why this can't be `const auto&`? (I guess it might be `const` implicitly since `container` is `const` but it can't hurt)

Reply:
Done